flake.lock: Update

max-privatevoid opened the pull request:
Flake lock file updates:

• Updated input 'flake-parts':
    'github:hercules-ci/flake-parts/9126214d0a59633752a136528f5f3b9aa8565b7d?narHash=sha256-sB4SWl2lX95bExY2gMFG5HIzvva5AVMJd4Igm%2BGpZNw%3D' (2024-04-01)
  → 'github:hercules-ci/flake-parts/506278e768c2a08bec68eb62932193e341f55c90?narHash=sha256-hgmguH29K2fvs9szpq2r3pz2/8cJd2LPS%2Bb4tfNFCwE%3D' (2024-11-01)
• Updated input 'nixpkgs':
    'github:NixOS/nixpkgs/7bb2ccd8cdc44c91edba16c48d2c8f331fb3d856?narHash=sha256-Drmja/f5MRHZCskS6mvzFqxEaZMeciScCTFxWVLqWEY%3D' (2024-04-25)
  → 'github:NixOS/nixpkgs/76612b17c0ce71689921ca12d9ffdc9c23ce40b2?narHash=sha256-IigrKK3vYRpUu%2BHEjPL/phrfh7Ox881er1UEsZvw9Q4%3D' (2024-11-09)

alecbcs commented:
@sug0 & @dukzcry thank you for your work fixing up Windows support and removing the Linux `ip` dependency! Apologies for the delay reviewing these PRs but this work looks really great!

alecbcs commented on tun/tun_windows.go:
```diff
@@ -0,0 +1,72 @@
+}
+
+// SetAddress sets the interface's known address and subnet.
+func SetAddress(name string, address string) (err error) {
```
@dukzcry is this function necessary within the Windows implementation or was it left in for compatibility? If this function successfully modifies the network is it possible to give water a bogus address space during creation and then use this to setup the real address?

dukzcry replied:
@alecbcs It is necessary. Unfortunately it is not viable to give water a bogus address at creation stage. Both are required to be set to correct address. You pass one at creation stage for TUN emulation to work properly in OpenVPN TAP driver and you set the other later like you set for any other interface. This is the limitation of their TUN implementation, please see https://github.com/songgao/water/blob/master/params_windows.go#L16 Maybe we should look at https://www.wintun.net/ later

alecbcs commented on tun/tun_windows.go:
```diff
@@ -0,0 +1,72 @@
+}
+
+// Delete removes a TUN device from the host.
+func Delete(name string) (err error) {
```
@dukzcry since this function simply returns `null`, how do Windows users remove/stop a running hyprspace instance?

dukzcry replied:
@alecbcs No additional step required. Once you close hyprspace, the interface goes down. This is specific to Windows implementation. You don't remove/create interfaces on Windows every time, just use the one provided and point hyprspace to it via InterfaceName.

alecbcs replied:
Okay interesting! So if hyprspace is running in the background we're going to need someone of killing it for the user which is how deleting the interface works in Linux. Maybe that means saving a record of the PID (or Windows equivalent) in an interface lock file and then killing that process when the user runs `hyprspace down`.

dukzcry replied:
@alecbcs Foreground mode never correctly worked for me at first. I will try to write a test case and open a new issue

dukzcry replied:
@alecbcs Ah, thanks! I didn't know that you can kill hyprspace running in background by issuing `hyprspace down`. I see it doesn't work indeed for Windows, the interface remains to be up and needs to be turned off manually.

sug0 commented:
@alecbcs I suggest you merge this PR into a separate branch, and work on potential changes to @dukzcry's Windows code from there.  I apologize for rebasing their commit into my PR, I made a mess. :P

alecbcs commented:
@sug0 no worries, I think I've got a fix in progress that's working for MacOS and Linux (I don't have a windows machine to test on)

alecbcs commented:
@sug0 @dukzcry if you two could both test the new version on main on Linux, Windows and let me know if you have any issues that would be great!

dukzcry commented:
@alecbcs Linux to Linux tunnel works fine
Windows doesn't build
```
tun/tun_windows.go:20:30: undefined: address
tun/tun_windows.go:71:9: undefined: errors
```

max-privatevoid committed:
flake.lock: Update

Flake lock file updates:

• Updated input 'flake-parts':
    'github:hercules-ci/flake-parts/9126214d0a59633752a136528f5f3b9aa8565b7d?narHash=sha256-sB4SWl2lX95bExY2gMFG5HIzvva5AVMJd4Igm%2BGpZNw%3D' (2024-04-01)
  → 'github:hercules-ci/flake-parts/506278e768c2a08bec68eb62932193e341f55c90?narHash=sha256-hgmguH29K2fvs9szpq2r3pz2/8cJd2LPS%2Bb4tfNFCwE%3D' (2024-11-01)
• Updated input 'nixpkgs':
    'github:NixOS/nixpkgs/7bb2ccd8cdc44c91edba16c48d2c8f331fb3d856?narHash=sha256-Drmja/f5MRHZCskS6mvzFqxEaZMeciScCTFxWVLqWEY%3D' (2024-04-25)
  → 'github:NixOS/nixpkgs/76612b17c0ce71689921ca12d9ffdc9c23ce40b2?narHash=sha256-IigrKK3vYRpUu%2BHEjPL/phrfh7Ox881er1UEsZvw9Q4%3D' (2024-11-09)
